pkg/operator/vsphereproblemdetector/vsphere_problem_detector_starter.go

```diff
@@ -0,0 +1,124 @@
+package vsphereproblemdetector
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+	openshiftv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/generated"
+	"github.com/openshift/cluster-storage-operator/pkg/operatorclient"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/controller/manager"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"k8s.io/klog/v2"
+)
+
+const (
+	infraConfigName = "cluster"
+)
+
+type VSphereProblemDetectorStarter struct {
+	controller     manager.ControllerManager
+	operatorClient *operatorclient.OperatorClient
+	infraLister    openshiftv1.InfrastructureLister
+	versionGetter  status.VersionGetter
+	targetVersion  string
+	eventRecorder  events.Recorder
+}
+
+func NewVSphereProblemDetectorStarter(
```

Comment:
I really don't like that we have 2 "starter" controllers whose main job is the same: start certain sub-controllers per platform.

I'd prefer to make the necessary adjustments to `CSIDriverStarterController` (to make it easily to start non-CSI controllers), rather than creating another starter controller.

Reply:
not that I disagree to it but can we do this in a future PR(I am happy to open a BZ/github issue to remind myself), because I think that this PR is quite large already.

Reply:
idea: when we have vSphere CSI driver, we can add detector deployment at an extra controller started with CSI driver, similarly as Manila starts certificate syncer:
https://github.com/openshift/cluster-storage-operator/blob/e7919b1ea165edd49142b783d99058fe65cc4f6a/pkg/operator/csidriveroperator/csioperatorclient/manila.go#L46-L48

It will be somewhat hidden in the CSI driver deployment, but still, it will run.

Reply:
I like that. Can we add a TODO entry in this starter controller?